src/main/java/fi/aalto/cs/apluscourses/intellij/model/CourseProject.java

```diff
@@ -107,7 +111,12 @@ public void removePasswordFromStorage(@NotNull PasswordStorage.Factory passwordS
 
   @NotNull
   public String getUserName() {
-    return user == null ? "" : user.getUserName();
+    return user.get() == null ? "" : Objects.requireNonNull(user.get()).getUserName();
```

Comment:
Either
```
var myUser = user.get(); 
return myUser == null ? "" : myUser.getUserName();
```
or
```
return Optional.ofNullable(user.get()).map(User::getUserName).orElse("")
```
The current code has possibility that `user.set(...)` is called between the conditional and the latter expression.